src/detect-bytejump.c

```diff
@@ -109,6 +136,20 @@ int DetectBytejumpDoMatch(DetectEngineThreadCtx *det_ctx, const Signature *s,
         SCReturnInt(0);
     }
 
+    /* Validate the number of bytes we are testing
+     * If the validation is successful, we know that
+     * it contains a value <= 23. Thus, we can
+     * safely cast it when extracting bytes
+     */
+    if (data->flags & DETECT_BYTEJUMP_NBYTES_VAR) {
+        if (!DetectBytejumpValidateNbytes(data, nbytes)) {
```

Comment:
If `DetectBytejumpValidateNbytes` can fail in this path, it will generate errors. If it can't, the check can be removed.

Reply:
I can refactor the validate function to only emit an error message when called as part of rule setup.

I think the validation should remain when a variable is used though.

Thoughts @victorjulien ?

Reply:
Works for me. As long as we can't get in a path with per packet errors I'm good.